spec/livefeed/parser_spec.rb

```diff
@@ -0,0 +1,320 @@
+          subject { post.tags }
+          it { is_expected.to eq([]) }
+        end
+      end
+    end
+
+    context "with a valid result containing outlinks" do
+      let(:fixture) { :valid_outlinks }
+
+      describe "#posts[0]" do
+        subject(:post) { result.posts[0] }
+        let(:expected_outlinks) do
+          %w[
+            http://softpir.ucoz.com/go?http://turbobit.net/4j7k3jn5jj6w.html
+            http://softpir.ucoz.com/go?http://uploadboy.me/dt7qllm4af3a.htm
+            http://softpir.ucoz.com/go?http://katfile.com/1g6l3x1oe738.htm
+            http://softpir.ucoz.com/go?http://www.file-upload.cc/7nfufe4nznw0
+            http://softpir.ucoz.com/news/2016-12-23-28176
+          ]
+        end
+
+        describe "#outlinks" do
+          subject { post.outlinks }
+          it { is_expected.to eq(expected_outlinks) }
+        end
+      end
+
+      describe "#posts[1]" do
+        subject(:post) { result.posts[1] }
+
+        describe "#outlinks" do
+          subject { post.outlinks }
+          it { is_expected.to eq(["http://www.dlvr.it/My4X64"]) }
+        end
+      end
+
+      describe "#posts[2]" do
+        subject(:post) { result.posts[2] }
+
+        describe "#outlinks" do
+          subject { post.outlinks }
+          it { is_expected.to eq([]) }
+        end
+      end
+    end
+
+    context "with a valid result containing non-blogs" do
+      let(:fixture) { :valid_non_blog }
+
+      it "should exclude the non-blog entries" do
+        expect(subject.posts.size).to eq(1)
+      end
+    end
+
+    context "with a valid empty result" do
+      let(:fixture) { :valid_empty }
+
+      it "should return an empty result" do
+        expect(subject.posts.size).to eq(0)
+        expect(subject.number_of_matches_total).to eq(0)
+        expect(subject.number_of_matches_returned).to eq(0)
+      end
+    end
+=end
+
+    context "with an unauthorized api key result" do
+      let(:fixture) { :unauthorized_api_key }
+
+      it "should raise AuthError" do
+        expect { subject }.to raise_error(LiveFeed::AuthError)
+      end
+    end
+
+    context "with an empty api key result" do
+      let(:fixture) { :empty_api_key }
+
+      it "should raise AuthError" do
+        expect { subject }.to raise_error(LiveFeed::QueryError)
+      end
+    end
+
+    context "with a 404 not found result" do
+      let(:fixture) { :not_found }
+
+      it "should raise QueryError" do
+        expect { subject }.to raise_error(LiveFeed::QueryError)
+      end
+    end
+
+    context "with a service unavailable result" do
+      let(:fixture) { :service_unavailable }
+
+      it "should raise ServerError" do
+        expect { subject }.to raise_error(LiveFeed::ServerError)
+      end
+    end
+
+    context "with a non XML response" do
+      let(:fixture) { :non_xml }
+
+      it "should raise ServerError" do
+        expect { subject }.to raise_error(LiveFeed::ServerError, /This should never happen/)
```

Comment:
> I was curious about the text here, "This should never happen", because it's an odd thing to write when dealing with computers. :-)

Hehe, true. I'll try to make it a bit more clear what happens :)

> Maybe a generic error message is better. (Or is useful error messages going to appear this way?)

Im not sure any useful message is going to appear this way so a generic error message would probably be better. This test was more or less copied from [the Search specs](https://github.com/twingly/twingly-search-api-ruby/blob/c9ed8d93b420fc672a647f46221cd9f9a6a0fba2/spec/parser_spec.rb#L298-L304), but the new version of LiveFeed API handles server errors better.